src/OrchardCore.Modules/OrchardCore.Users/Liquid/HasPermissionFilter.cs

```diff
@@ -30,12 +30,19 @@ public async ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArgument
             var permissionName = arguments["permission"].Or(arguments.At(0)).ToStringValue();
             var resource = arguments["resource"].Or(arguments.At(1)).ToObjectValue();
 
-            var permission = permissionProviders.SelectMany(pp => pp.GetPermissions().Where(p => p.Name == permissionName))
-                                .Where(c => c != null).FirstOrDefault();
+            Permission permission = null;
```

Comment:
That's a weird one, why not `foreach` ?

Reply:
sorry, no reason. foreach works best over IEnumerable<>